src/rest_api/classes/person/widgets/publications.clj

```diff
@@ -0,0 +1,71 @@
+(ns rest-api.classes.person.widgets.publications
+  (:require
+   [clojure.string :as str]
+   [datomic.api :as d]
+   [pseudoace.utils :as pace-utils]))
+
+
+(defn publications [person]
+   (let [db (d/entity-db person)
+         dataArrayTuples
+         (->> (d/q '[:find [?paper ...]
+                   :in $ ?person
+                   :where [?paperperson :paper.person/person ?person]
+                          [?paper :paper/person ?paperperson]
+                          [?paper :paper/type ?typeent]
+                          (not [?typeent :paper.type/type :paper.type.type/meeting-abstract])
+                          ]
+                 db (:db/id person))
+              (map (fn [oid]
+                     (let [paper (d/entity db oid)]
+                      (pace-utils/vmap
+                       :year (first (str/split (:paper/publication-date paper) #"-"))
+                       :object
+                        (pace-utils/vmap
+                         :taxonomy "all"
+                         :class "paper"
+                         :label (:paper/brief-citation paper)
+                         :id (:paper/id paper))
+                       :brief_citation (:paper/brief-citation paper)
+                     ))))
+              (seq))]
+     (def data (group-by :year dataArrayTuples))
```

Comment:
You can eliminate the `eastwood` warnings by removing this line (32),
and adding `((group-by :year))` as the final step in the `->>` pipeline, just below `(seq)` above.